src/index.js

```diff
@@ -1,21 +1,32 @@
-import React, { Component } from 'react';
+import React, { PureComponent } from 'react';
 import PropTypes from 'prop-types';
 
 import Animate from './animate.custom.css';
 import styles from './styles.css';
 
-export default class StepWizard extends Component {
-    constructor(props) {
-        super(props);
+export default class StepWizard extends PureComponent {
+    state = {
+        activeStep: 0,
+        classes: {},
+        hashKeys: {},
+    }
 
-        this.state = this.initialState();
+    componentDidMount() {
+        this.setState({ ...this.initialState() });
```

Comment:
Most of this looks great, but my concern is calling `setState` inside `componentDidMount` since it will render twice on load. What are your thoughts?

Reply:
It doesn't render twice on load. It's not a  anti-pattern because you can call setState when you use some async operations like fetch some values of api (https://reactjs.org/docs/faq-ajax.html).
If component calls this lifecycle again then it should destroy all values and remove subscribers. I think if it is passed into a shouldComponentUpdate or componentDidUpdate should generate errors like what you said.

Reply:
This is in the React [documentation](https://reactjs.org/docs/react-component.html#componentdidmount):
`You may call setState() immediately in componentDidMount(). It will trigger an extra rendering, but it will happen before the browser updates the screen.`

It's definitely not an anti-pattern but I wanted to consider any potential performance issues. We don't know what kind of components with lifecycles people will use as steps so if they are using `componentDidMount` too the initial state won't be set in the first, unnoticeable, render.

If you don't think this should be an issue I'll go ahead and merge.

Reply:
I'll try to change it. Thank you for the explanation :smile: